addons/l10n_th/models/res_partner.py

```diff
@@ -0,0 +1,10 @@
+from odoo import models
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _l10n_th_get_branch_name(self):
+        if not self.is_company:
+            return ""
+        code = self.company_registry or ""
+        return "Branch " + code if code else "Headquarter"
```

Comment:
```suggestion
        code = self.company_registry or "Headquarter"
        return f"Branch {code}"
```
More clearer like that!

Reply:
The branch naming will be either "Headquarter" or "Branch {branch_number}"
So I think I will do:
```
code = self.company_registry
return f"Branch {code}" if code else "Headquarter"
```